beacon-chain/blockchain/execution_engine.go

```diff
@@ -61,18 +59,12 @@ func (s *Service) notifyForkchoiceUpdate(ctx context.Context, arg *notifyForkcho
 	if err != nil {
 		return nil, errors.Wrap(err, "could not get execution payload")
 	}
-	finalizedHash, err := s.getPayloadHash(ctx, arg.finalizedRoot)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not get finalized block hash")
-	}
-	justifiedHash, err := s.getPayloadHash(ctx, arg.justifiedRoot)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not get justified block hash")
-	}
+	finalizedHash := s.store.FinalizedPayloadBlockHash()
+	justifiedHash := s.store.JustifiedPayloadBlockHash()
```

Comment:
Use cached payload hashes instead of retrieving it from DB from given input roots